staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -1581,3 +1585,219 @@ func TestCacheIntervalInvalidationStopsWatch(t *testing.T) {
 		t.Errorf("unexpected number of events received, expected: %d, got: %d", bufferSize+1, received)
 	}
 }
+
+func TestBookmarkAfterResourceVersionWatchers(t *testing.T) {
+	newWatcher := func(id string, deadline time.Time) *cacheWatcher {
+		return newCacheWatcher(0, func(_ string, _ labels.Set, _ fields.Set) bool { return true }, func(bool) {}, testVersioner{}, deadline, true, objectType, id, 10)
+	}
+
+	clock := testingclock.NewFakeClock(time.Now())
+	target := newTimeBucketWatchers(clock, defaultBookmarkFrequency)
+	target.addWatcher(newWatcher("1", clock.Now().Add(2*time.Minute)))
+
+	// the watcher is immediately expired
```

Comment:
in this case, nextBookmarkTime returns `time.Time{}, true`

But then what happens we're doing:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L190

so we're effectively getting negative duration. This works but is a bit unexpected to me.
Maybe lets add a comment in the nextBookmarkTime method ardoun that line that we can actually get negative duration and that works fine and is expected.